src/Adapter/Product/ProductDataProvider.php

```diff
@@ -61,16 +61,11 @@ public function getProductInstance($idProduct = null)
      * @param int|null $id_shop
      * @param object|null $context
      *
-     * @throws \LogicException If the product id is not set
      *
      * @return Product $product
      */
     public function getProduct($id_product, $full = false, $id_lang = null, $id_shop = null, $context = null)
     {
-        if (!$id_product) {
-            throw new \LogicException('You need to provide a product id', 5002);
-        }
-
         $product = new Product($id_product, $full, $id_lang, $id_shop, $context);
```

Comment:
TLDR you need to find a place where it provides the "0" value as an id to that method, and make sure it still provides the valid product id taken from the order detail (then when you click on it, it should redirect to products list with a message that product was not found or so.. As usuall with the missing entities)

Reply:
As I see from this [message](https://github.com/PrestaShop/PrestaShop/issues/29913#issuecomment-1384209031) @MatShir should have the expected behavior ready somewhere, so maybe he can clarify

Reply:
>As I see from this https://github.com/PrestaShop/PrestaShop/issues/29913#issuecomment-1384209031 @MatShir should have the expected behavior ready somewhere, so maybe he can clarify

Hi @zuk3975, I already clarify the behaviour with @MatShir :)

>the method is used by other services and it is intended to fetch the product, so the exception makes sense here.
This is not the way to fix it.
We could ask more details from UX team of how it supposed to work, but for a quick fix it could at least behave same as in order view page (see screenshot - it warns that the product is out of stock (the product was actually deleted so the warning could be improved. And when you click on it just goes to the list of all products - IMHO there should also be a note that product was not found).. Anyway as I said we might need to consult UX of what behavior we expect in such scenario and make it work the same in order preview and order view pages.

Actually if you see well, there is a problem in the design of this function. Indeed, unlike the php doc informations, the parameter `$id_product` is not always an int. `$id_product` can be int OR string OR null ... 
For exemple here a string is given to the `getProduct` method.
https://github.com/PrestaShop/PrestaShop/pull/30924/files#diff-15855f366567048d282cf3421256e194734dd21b17c9b47f48536474454fd6ecR463
I wonder why we check  `if(!product)` to throw the LogicException. 
1. If you say, "this is a check for valid object model id", I response: if `$id_product` = '0' we throw the exception but if `$id_product` = '-1' we don't. Very strange (By the way, I don't think we should test valid object model id outside the objet model).
2. If you DON'T say "this is a check for valid object model id", this my question: Why do we add this extra check `if(!product)` to throw exception only for 0, '0',  '' or null value? If `$id_product` is one of them the `new Product()` would be invalid the same way that if `$id_product` would be '15' but not existing in db (I mean not found).